common/models/role.js

```diff
@@ -87,6 +87,8 @@ module.exports = function(Role) {
           principalModel = this.constructor.registry.findModel(query.where.principalType);
           principalType = query.where.principalType;
         }
+        // make sure we don't keep principalType in userModel query
+        delete query.where.principalType;
 
```

Comment:
here i fix a missing part i forgot to port from my previous attempt with userModelName property

I admit this is a bit tricky since we use a principalType property in the Role.users() query to resolve the user model, then remove this property as it won't ever it in the user model (thus making [this request](https://github.com/strongloop/loopback/pull/2971/commits/a49ae19599c6ed09edd295f613fcf9dfed1e59d8#diff-733982793c3bd5e61d1a4350451fb494R128) fail). 

But this Role.user() if from starting point far from being conventional since it looks like a classic relation on Role model, but isn't at all underneath. This could be fixed using a polymorphic+through relation between role and user in a custom Role model but i'm not sure to feel the added value for now